sway-core/src/type_engine/engine.rs

```diff
@@ -47,6 +47,60 @@ impl Engine {
             (Ref(received), _) => self.unify(received, expected, span),
             (_, Ref(expected)) => self.unify(received, expected, span),
 
+            (Unknown, Numeric) => match self.slab.replace(
+                received,
+                &Unknown,
+                TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
+            ) {
+                None => {
+                    match self.slab.replace(
+                        expected,
+                        &Numeric,
+                        TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
+                    ) {
+                        None => Ok(vec![]),
+                        Some(_) => self.unify(received, expected, span),
+                    }
+                }
+                Some(_) => self.unify(received, expected, span),
+            },
+
+            (Numeric, Unknown) => match self.slab.replace(
+                expected,
+                &Unknown,
+                TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
+            ) {
+                None => {
+                    match self.slab.replace(
+                        received,
+                        &Numeric,
+                        TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
+                    ) {
+                        None => Ok(vec![]),
+                        Some(_) => self.unify(received, expected, span),
+                    }
+                }
+                Some(_) => self.unify(received, expected, span),
+            },
+
+            (Numeric, Numeric) => match self.slab.replace(
```

Comment:
Is it necessary for two numerics to resolve to a `u64`? why not stay numeric?

Reply:
Fixed.